stage1/init/kvm/resources.go

```diff
@@ -68,3 +68,17 @@ func GetAppsResources(apps schema.AppList) (totalCpus, totalMem int64) {
 
 	return totalCpus, totalMem
 }
+
+// GetAppCapabilities is a filter which returns list of valid Linux capabilities
+// Function requires a manifest of application
+func GetAppCapabilities(app *types.App) []types.LinuxCapability {
+
+	var caps []types.LinuxCapability
+
+	for _, isolator := range app.Isolators {
+		if capSet, ok := isolator.Value().(types.LinuxCapabilitiesSet); ok {
```

Comment:
Checking the interface `types.LinuxCapabilitiesSet` means it will take [both isolators related to caps](https://github.com/appc/spec/blob/master/spec/ace.md#linux-isolators):
- os/linux/capabilities-remove-set
- os/linux/capabilities-retain-set

There is a similar loop in [appToNspawnArgs](https://github.com/coreos/rkt/blob/master/stage1/init/common/pod.go#L696) and it is using the retain set.